### DISCUSSION TITLE
Can someone teach me how to use GestureRecognizers?

### DISCUSSION THREAD

**Discussion by PsychedelicBeans** [AvaloniaUI/Avalonia#13842] [Q&A]
I've been trying to use the ScrollGestureRecognizer for a while now, but I must be stupid, because I can't figure it out, and the documentation is no help. 

As a test, I've got a simple Rectangle with a ScrollGestureRecognizer attached

<Rectangle Name="Rect" Width="100" Height="100" Fill="Red" VerticalAlignment="Center" HorizontalAlignment="Center">
	<Rectangle.GestureRecognizers>
		<ScrollGestureRecognizer x:Name="RectScrollGesture" CanHorizontallyScroll="True" CanVerticallyScroll="True" />
	</Rectangle.GestureRecognizers>
</Rectangle>

The way I expect it to work is by accessing either the recognizer itself or its parent item, the rectangle, and listening to an OnScroll event, but that doesn't seem to be an option. In fact, I don't see any events being raised in the source code for the scroll gesture recognizer.

What am I doing wrong?

**Comment by thevortexcloud** [comment]
Does any of this help?

https://github.com/AvaloniaUI/Avalonia/blob/master/samples/ControlCatalog/Pages/GesturePage.xaml
https://github.com/AvaloniaUI/Avalonia/blob/master/samples/ControlCatalog/Pages/GesturePage.cs

I have never used it myself so those samples are really the best I can do.
